extern/iso_c_fortran_bmi/test/test_iso_c.c

```diff
@@ -0,0 +1,327 @@
+#include <dlfcn.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+extern register_bmi(void *);
+extern initialize(void *, char *);
+extern update(void *);
+extern update_until(void*, double *);
+extern finalize(void*);
+extern get_component_name(void *, char *);
+extern get_input_item_count(void*, int *);
+extern get_output_item_count(void*, int *);
+extern get_input_var_names(void*, char**);
+extern get_output_var_names(void*, char**);
+extern get_var_grid(void*, char*, int*);
+extern get_var_type(void*, char*, char*);
+extern get_var_units(void*, char*, char*);
+extern get_var_itemsize(void*, char*, int*);
+extern get_var_nbytes(void*, char*, int*);
+extern get_var_location(void*, char*, char*);
+extern get_current_time(void*, double *);
+extern get_start_time(void*, double *);
+extern get_end_time(void*, double *);
+extern get_time_units(void*, char *);
+extern get_time_step(void*, double *);
+extern get_value_int(void*, char*, int*);
+extern get_value_float(void*, char*, float*);
+extern get_value_double(void*, char*, double*);
+extern set_value_int(void*, char*, int*);
+extern set_value_float(void*, char*, float*);
+extern set_value_double(void*, char*, double*);
+extern get_grid_rank(void*, int*, int*);
+extern get_grid_size(void*, int*, int*);
+extern get_grid_type(void*, int*, char*);
+extern get_grid_shape(void*, int*, int*);
+extern get_grid_spacing(void*, int*, double*);
+extern get_grid_origin(void*, int*, double*);
+extern get_grid_x(void*, int*, double*);
+extern get_grid_y(void*, int*, double*);
+extern get_grid_z(void*, int*, double*);
+extern get_grid_node_count(void*, int*, int*);
+extern get_grid_edge_count(void*, int*, int*);
+extern get_grid_face_count(void*, int*, int*);
+extern get_grid_edge_nodes(void*, int*, int*);
+extern get_grid_face_edges(void*, int*, int*);
+extern get_grid_face_nodes(void*, int*, int*);
+extern get_grid_nodes_per_face(void*, int*, int*);
```

Comment:
possibly move these to a header file.

Reply:
In the core NGEN framework code, they are in a header, this was just meant to be a standalone C/Fortran test to exercise the functions.  Not sure it needs much attention beyond that.